source/kraken/realtime.cpp

```diff
@@ -528,31 +564,21 @@ void handle_realtime(const std::string& id,
     auto log = log4cplus::Logger::getInstance("realtime");
     LOG4CPLUS_TRACE(log, "realtime trip update received");
 
-    if (! is_handleable(trip_update)
+    if (! is_handleable(trip_update, *data.pt_data, is_realtime_add_enabled, is_realtime_add_trip_enabled)
             || ! check_trip_update(trip_update)) {
         LOG4CPLUS_DEBUG(log, "unhandled real time message");
         return;
     }
-    // if is_realtime_add_enabled = false, this disables the adding trip option too
-    if ((! is_realtime_add_enabled && is_added_stop_time(trip_update)) ||
-        (! is_realtime_add_enabled && is_added_trip(trip_update))) {
-        LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring trip update id "
-                << trip_update.trip().trip_id()
-                << " with ADDED/ADDED_FOR_DETOUR stop times or ADDITIONAL_SERVICE effect");
-        return;
-    }
-    if (! is_realtime_add_trip_enabled && is_added_trip(trip_update)) {
-        LOG4CPLUS_DEBUG(log, "Add trip is disabled: ignoring trip update id "
-                << trip_update.trip().trip_id()
-                << " with ADDITIONAL_SERVICE effect");
-        return;
-    }
 
-    auto meta_vj = data.pt_data->meta_vjs.get_mut(trip_update.trip().trip_id());
-    if (! meta_vj) {
-        LOG4CPLUS_ERROR(log, "unknown vehicle journey " << trip_update.trip().trip_id());
-        // TODO for trip().ADDED, we'll need to create a new VJ
-        return;
+    // Get or create meta VJ
+    navitia::type::MetaVehicleJourney* meta_vj = nullptr;
+    if (! data.pt_data->meta_vjs.exists(trip_update.trip().trip_id())) {
```

Comment:
We add it only if it's an ADDED trip, I think.
Otherwise it should be rejected.

Reply:
Good idea. I think it is preferable too